utils/pre-publish-check.js

```diff
@@ -16,39 +18,55 @@ glob(
             const {publishConfig, name} = pkgJson;
 
             if (!publishConfig || publishConfig.access !== "public") {
-                // eslint-disable-next-line no-console
                 console.error(
                     `ERROR: ${name} is missing a "publishConfig": {"access": "public"} section.`,
                 );
                 process.exit(1);
             }
 
+            if (pkgJson.main !== "dist/index.js") {
+                console.error(
+                    `ERROR: ${name} must have a "main" set to "dist/index.js".`,
+                );
+                process.exit(1);
+            }
+
             if (pkgJson.private) {
-                // eslint-disable-next-line no-console
                 console.warn(
                     `${name} is private and won't be published to NPM.`,
                 );
                 warnings = true;
             }
         }
 
-        if (warnings) {
-            inquirer
-                .prompt([
-                    {
-                        type: "confirm",
-                        name: "skipWarnings",
-                        default: false,
-                        message:
-                            "There are some potential problems, do you wish to continue?",
-                    },
-                ])
-                .then(({skipWarnings}) => {
-                    // If we're not skipping then we need to exit with an error
-                    if (!skipWarnings) {
-                        process.exit(1);
-                    }
-                });
-        }
+        exec("npm whoami", (err, currentUser) => {
+            if (currentUser.trim() !== "khanacademy") {
+                console.error(
+                    `ERROR: You are not logged in to NPM as "khanacademy". ` +
+                        `Run "npm login" and use the password from: ` +
+                        `https://phabricator.khanacademy.org/K207`,
+                );
+                process.exit(1);
+            }
+
+            if (warnings) {
+                inquirer
+                    .prompt([
+                        {
+                            type: "confirm",
+                            name: "skipWarnings",
+                            default: false,
+                            message:
+                                "There are some potential problems, do you wish to continue?",
+                        },
+                    ])
+                    .then(({skipWarnings}) => {
+                        // If we're not skipping then we need to exit with an error
+                        if (!skipWarnings) {
+                            process.exit(1);
+                        }
+                    });
+            }
+        });
```

Comment:
Perhaps this whole condition should be in a method like `_maybeContinueAfterWarnings()`.

Would've made diff a little cleaner as that big chunk of red on the left and this green would've more easily correlated.